BUILD.sh

```diff
@@ -1,69 +1,226 @@
 #!/bin/bash
 
-set -x
+cleanup() {
+  echo
+  echo Exit code $1 caught in build script.
+  echo
+  if [[ "$1" == "0" ]]; then
+    echo BUILD_STATUS:0
+  else
+    echo
+    echo Failure found on line $2 in build script.
+    echo
+    echo BUILD_STATUS:1
+  fi
+}
 
-#  NOTE: The following is required when running from Gitlab CI via slurm job
-source /etc/profile.d/modules.sh
+trap 'cleanup $? $LINENO' EXIT
 
-export MY_CLUSTER=`uname -n | sed -e 's/[0-9]//g' -e 's/\..*//'`
-export NVBLAS_CONFIG_FILE=$(pwd)/nvblas.conf
+export BUILD=1
+export TEST=1
+while [[ $# -gt 0 ]]
+do
+  case $1 in
+    --build-only|-B)
+      echo
+      echo Building only
+      echo
+      export BUILD=1
+      export TEST=0
+      shift
+      ;;
+    --test-only|-T)
+      echo
+      echo Testing only
+      echo
+      export BUILD=0
+      export TEST=1
+      shift
+      ;;
+    *)
+      cat <<EOD
+    Argument not found!
+      
+    usage: $0 [ --test-only|-T ] [ --build-only|-B ]
+EOD
+      exit 1
+      ;;
+  esac
+done
+
+set -x
+x="unset"
+
+if [[ ! -v MY_CLUSTER ]]
+then
+  export MY_CLUSTER=`uname -n | sed -e 's/[0-9]//g' -e 's/\..*//'`
```

Comment:
`uname -n` returns "login1" on ascent :(

Reply:
I usually pass `MY_CLUSTER=ascent ./BUILD.sh` since that sysname discovery can be quite misleading (this is how we run in CI as well). I question whether we should use that at all or require users to specify which cluster they are running on.

Reply:
I'll put a note at the top of the script file

Reply:
follow-up: I get this error when I ran BUILD.sh on  ascent

```
jsrun -n 1 -g 1 /ccsopen/home/petra1/hiop/build/src/Drivers/nlpMDS_ex4_raja.exe 400 100 1 -selfcheck
Error: It is only possible to use js commands within a job allocation unless CSM is running
10-16-2020 17:05:21:448 95719 main: Error initializing RM connection. Exiting.
```